javascript/extractor/src/com/semmle/js/extractor/ASTExtractor.java

```diff
@@ -786,11 +838,31 @@ public Label visit(AssignmentExpression nd, Context c) {
       return key;
     }
 
+    // set to determine which BinaryExpression has been extracted as regexp
+    private Set<Expression> extractedAsRegexp = new HashSet<>();
```

Comment:
This seems a bit heavy-handed for detecting the root BinaryExpression. Ideally this should be part of the `Context` class.

If for some reason you'd rather use the set, I'd suggest
- Use a name like `shouldNotExtractAsRegExp` or `parentWillExtractAsRegExp` to make it clear how it is used.
- Remove nodes again when they have been visited.